### PR TITLE
Update update-deps.yml with dockerfile update

### PR DESCRIPTION
**Pull request by jingxu97**
This change set up a job to update go version in dockerfile. This only updates 1.22 patch version.

### CODE REVIEW

File: .github/workflows/update-deps.yml [hunk 1]
```diff
@@ -43,3 +43,38 @@ jobs:
           labels: ok-to-test
           body: |
             Updating go.mod with latest dependencies...
+  update-go-version:
```
Comment (wangzhen127) [inline .github/workflows/update-deps.yml:46]: Does this need to be a separate job?

**Review by wangzhen127 — COMMENTED** [review]
> This only updates 1.22 patch version.

Does this mean golang minor version update is still manual? And when that happens, we also need to change update-deps.yml together?

File: .github/workflows/update-deps.yml (hunk 1 above)
Comment (jingxu97) [reply .github/workflows/update-deps.yml:46]: maybe not, but separate out to avoid one failure affect the other?

**Comment by jingxu97** [comment]
> > This only updates 1.22 patch version.
> 
> Does this mean golang minor version update is still manual? And when that happens, we also need to change update-deps.yml together?

Yes for now. But minor version update is about once a year. So the overhead is small.

**Pull request merged**